File: Modules/Membership/MyProfile/MyProfileForm.cs
using Serenity.ComponentModel;
using Serenity.Data.Mapping;
using Serenity.Web;
using System;

namespace WinAuthSample.Membership.Forms
{
    [FormScript("Membership.MyProfile")]
    [BasedOnRow(typeof(MyProfileRow), CheckNames = true)]
    public class MyProfileForm
    {

        
        //public string Username { get; set; }
        [LabelWidth(200, UntilNext = true)]

        public string DisplayName { get; set; }

        [Size(4), MaxLength(4)]
        public string FormInitials { get; set; }


       // [TwoThirdWidth]
        public string Email { get; set; }

        //[OneThirdWidth]
        //public string MobilePhoneNumber { get; set; }
      

        /*
        TwoThirdWidth]
        public string SignatureName { get; set; }
        */
        

        /*
        [OneThirdWidth]
        public string SignatureTitle { get; set; }
        */
       
        

        [TwoThirdWidth]
        public string UserImage { get; set; }

        [Hidden]
        public string Username { get; set; }
     
     
   
    }
}

File: Modules/ServerTypes/Membership/MyProfileForm.ts
import { StringEditor, EmailAddressEditor, ImageUploadEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface MyProfileForm {
    DisplayName: StringEditor;
    FormInitials: StringEditor;
    Email: EmailAddressEditor;
    UserImage: ImageUploadEditor;
    Username: StringEditor;
}

export class MyProfileForm extends PrefixedContext {
    static readonly formKey = 'Membership.MyProfile';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!MyProfileForm.init)  {
            MyProfileForm.init = true;

            var w0 = StringEditor;
            var w1 = EmailAddressEditor;
            var w2 = ImageUploadEditor;

            initFormType(MyProfileForm, [
                'DisplayName', w0,
                'FormInitials', w0,
                'Email', w1,
                'UserImage', w2,
                'Username', w0
            ]);
        }
    }
}